packages/api-client/src/bots/BotsAPI.ts

```diff
@@ -0,0 +1,215 @@
+   */
+  public async getConversation(userId: string): Promise<ClientSeenByBot[]> {
+    const config: AxiosRequestConfig = {
+      method: 'get',
+      url: `${BotsAPI.URL.BOT}/${BotsAPI.URL.USERS}/${userId}${BotsAPI.URL.CLIENTS}`,
+    };
+
+    const response = await this.client.sendJSON<ClientSeenByBot[]>(config);
+    return response.data;
+  }
+
+  /**
+   * Fetch the bot's own user profile information.
+   */
+  public async getSelf(): Promise<Self> {
+    const config: AxiosRequestConfig = {
+      method: 'get',
+      url: `${BotsAPI.URL.BOT}/${BotsAPI.URL.SELF}`,
+    };
+
+    const response = await this.client.sendJSON<Self>(config);
+    return response.data;
+  }
+
+  /**
+   * Claim prekeys from one or more clients of one or more users.
+   *
+   * The result is a nested JSON object structure where the first-level keys
+   * are user IDs and the second-level keys client IDs. The values paired with
+   * the client IDs are the base64-encoded prekeys.
+   */
+  public async postUserPreKeysList(userPreKeyList: UserPreKeyList): Promise<UserPreKeyDataList> {
+    const config: AxiosRequestConfig = {
+      data: userPreKeyList,
+      method: 'post',
+      url: `${BotsAPI.URL.BOT}/${BotsAPI.URL.CLIENT}/${BotsAPI.URL.PREKEYS}`,
+    };
+
+    const response = await this.client.sendJSON<UserPreKeyDataList>(config);
+    return response.data;
+  }
+
+  /**
+   * Fetch other user's profiles.
+   */
+  public async getUsers(ids: string[]): Promise<UserSeenByBot[]> {
+    const config: AxiosRequestConfig = {
+      method: 'get',
+      url: `${BotsAPI.URL.BOT}/${BotsAPI.URL.USERS}`,
+    };
+
+    const response = await this.client.sendJSON<UserSeenByBot[]>(config);
+    return response.data;
+  }
+
+  /**
+   * List the bot's remaining prekey IDs.
+   */
+  public async postClientPreKeys(preKeys: PreKey[]): Promise<void> {
+    const config: AxiosRequestConfig = {
+      data: {prekeys: preKeys},
+      method: 'post',
+      url: `${BotsAPI.URL.BOT}/${BotsAPI.URL.CLIENT}/${BotsAPI.URL.PREKEYS}`,
+    };
+
+    await this.client.sendJSON(config);
+  }
+
+  /**
+   * Upload an asset.
+   * Note that resumable uploads are not currently supported for bots.
+   */
+  public postAsset(
+    asset: Uint8Array,
+    options?: AssetOptions,
+    progressCallback?: ProgressCallback,
+  ): Promise<RequestCancelable<AssetUploadData>> {
+    const customUrl = `${BotsAPI.URL.BOT}/${BotsAPI.URL.ASSETS}`;
+    return new AssetAPI(this.client).postAsset(asset, options, progressCallback, customUrl);
+  }
+
+  /**
+   * Download an asset.
+   */
+  public getAsset(
+    assetId: string,
+    token?: string | null,
+    forceCaching: boolean = false,
+    progressCallback?: ProgressCallback,
+  ): Promise<RequestCancelable<AssetResponse>> {
+    const customUrl = `${BotsAPI.URL.BOT}/${BotsAPI.URL.ASSETS}/${assetId}`;
+    return new AssetAPI(this.client).getAssetV3(assetId, token, forceCaching, progressCallback, customUrl);
```

Comment:
I think it would be better to pass an AssetAPI instance in the constructor instead of creating new ones here